aptos-move/e2e-tests/src/executor.rs

```diff
@@ -736,40 +823,98 @@ impl<O: OutputLogger> FakeExecutorImpl<O> {
     fn execute_transaction_block_impl_with_state_view(
         &self,
         txn_block: Vec<SignatureVerifiedTransaction>,
-        onchain_config: BlockExecutorConfigFromOnchain,
-        sequential: bool,
         state_view: &(impl StateView + Sync),
+        config: BlockExecutorConfig,
+        metadata: Option<TransactionSliceMetadata>,
     ) -> Result<Vec<TransactionOutput>, VMStatus> {
-        let config = BlockExecutorConfig {
-            local: BlockExecutorLocalConfig {
-                blockstm_v2: false,
-                concurrency_level: if sequential {
-                    1
-                } else {
-                    usize::min(4, num_cpus::get())
-                },
-                allow_fallback: self.allow_block_executor_fallback,
-                discard_failed_blocks: false,
-                module_cache_config: BlockExecutorModuleCacheLocalConfig::default(),
-            },
-            onchain: onchain_config,
-        };
         let txn_provider = DefaultTxnProvider::new_without_info(txn_block);
-        AptosVMBlockExecutorWrapper::execute_block_on_thread_pool::<
-            _,
-            NoOpTransactionCommitHook<VMStatus>,
-            _,
-        >(
-            self.executor_thread_pool.clone(),
-            &txn_provider,
-            &state_view,
-            // Do not use shared module caches in tests.
-            &AptosModuleCacheManager::new(),
-            config,
-            TransactionSliceMetadata::unknown(),
-            None,
-        )
-        .map(BlockOutput::into_transaction_outputs_forced)
+        let requested = txn_provider.num_txns();
+        let result = {
+            AptosVMBlockExecutorWrapper::execute_block_on_thread_pool::<
+                _,
+                NoOpTransactionCommitHook<VMStatus>,
+                _,
+            >(
+                self.executor_thread_pool.clone(),
+                &txn_provider,
+                &state_view,
+                self.module_cache_manager_opt()
+                    .unwrap_or(&AptosModuleCacheManager::new()),
+                config,
+                metadata.unwrap_or_else(TransactionSliceMetadata::unknown),
+                None,
+            )
+            .map(BlockOutput::into_transaction_outputs_forced)
+        };
+        let mut outputs = result?;
+        // Trim any appended BlockEpilogue output to preserve empty metadata behavior.
```

Comment:
I am still concerned about this change, why do we need to trim output? How come tests were okay before without this change, cc @grao1991 - it seems like we have a mismatch in block size now?

Reply:
I see, great, so no bugs here?